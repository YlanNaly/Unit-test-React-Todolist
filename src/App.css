ul {
    text-decoration: none;
    list-style-type: none;
}
.container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}
.App{
    display: flex;
    flex-direction: row;
    gap: 12px;
}
.todo {
    display: flex;
    max-width: 520px;
    text-align: center;
    margin-top: 12px;
    max-height:520px;
    overflow: auto;
    scroll-behavior: smooth;
    flex-direction: column;
}
.done {
    float: left;
    margin-top: 12px;
    max-width: 520px;
    max-height:520px;
    overflow: auto;
    scroll-behavior: smooth;
    text-align: center;
    flex-direction: column;
}