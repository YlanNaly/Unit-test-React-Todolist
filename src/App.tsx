import {useState} from "react";
import AddTask from "./todolist/Ajout";
import Todo from "./todolist/Todo";
import DONE from "./todolist/Done";

function App() {

    const [tasks, setTasks] = useState([
        { id: 1, text: "Manger!", done: true },
    ]);
    const addTask = (text: any) => {
        const newTask = {
            text,
            id: Date.now(),
            done: false,
        };
        setTasks([...tasks, newTask]);
    };

    const toggleTask = (id:number ) => {
        const realTask = tasks.find((t: { id: number; }) => t.id === id);
        const index = tasks.findIndex((t: { id: number; }) => t.id === id);
        const taskCopy = { ...realTask };
        const tasksListCopy :any = [...tasks];

        tasksListCopy[index] = taskCopy;
        setTasks(tasksListCopy);
    };

    const [done, setDone] = useState([
        {}
    ]);
    const addToDone = (index: number) => {
        setDone([...done, tasks[index]]);
        const tmp = [...tasks];
        tmp.splice(index, 1);
        setTasks(tmp);
    }

    const deleteTask3 = (id: number) => {
        const tmp = [...done];
        tmp.splice(id, 1);
        setDone(tmp);
    }

    return (
        <>
        <AddTask addTask={addTask} />
        <div className="App">
            <div className="todo" style={{width: "18rem"}}>
                <div className="Todo-header">TO DO</div>
                <ul className="list-group list-group-flush">
                    <Todo tasks={tasks} toggleTask={toggleTask} deleteTask={addToDone}/>
                </ul>
            </div>
            <div className="done" style={{width: "18rem"}}>
                <div className="Done-header">DONE</div>
                <ul className="list-group list-group-flush">
                    <DONE tasks={done} toggleTask={toggleTask} deleteTask={deleteTask3}/>
                </ul>
            </div>
        </div></>


    );
}

export default App;


