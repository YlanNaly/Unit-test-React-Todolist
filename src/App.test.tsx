import {render} from "@testing-library/react";
import App from "./App";
import {fireEvent , screen} from "@testing-library/react";

test("Renders Table  - First Row Task", () => {
  render(<App />);
  const linkElement = screen.getByText("Manger!");
  expect(linkElement).toMatchSnapshot();
});