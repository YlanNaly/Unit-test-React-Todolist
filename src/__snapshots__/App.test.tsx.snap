// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Table  - First Row Task 1`] = `
<li
  class="list-group-item"
  data-testid="list"
>
  Manger!
  <span
    class="forme"
    data-testid="deleteTask"
    role="button"
    style="padding: 5px; margin-left: 20px; font-size: 20px;"
  >
    <input
      data-testid="x"
      type="checkbox"
    />
  </span>
</li>
`;
